Source/Rhetos.Deployment/ApplicationGenerator.cs

```diff
@@ -24,111 +24,124 @@
-            _logger.Info("Application model has " + dslModelConceptsCount + " statements.");
+            var additionalDependencies = ParseAdditionalDependencies();
+
+            var job = new ParallelTopologicalJob(_logProvider);
+            foreach (var generator in generators)
+            {
+                var dependencies = generator.Dependencies ?? Enumerable.Empty<string>();
+                var generatorName = GetGeneratorName(generator);
+                if (additionalDependencies.TryGetValue(generatorName, out var additionForGenerator))
+                    dependencies = dependencies.Concat(additionForGenerator);
+
+                job.AddTask(generatorName, () =>
+                {
+                    _logger.Info(() => $"Starting generator '{generatorName}'.");
+                    generator.Generate();
+                }, dependencies);
+            }
+
+            return job;
         }
 
-        private IList<IGenerator> GetSortedGenerators()
+        private Dictionary<string, List<string>> ParseAdditionalDependencies()
         {
-            // The plugins in the container are sorted by their dependencies defined in ExportMetadata attribute (static typed):
-            var generators = _generatorsContainer.GetPlugins().ToArray();
-
-            // Additional sorting by loosely-typed dependencies from the Dependencies property:
-            var generatorNames = generators.Select(GetGeneratorName).ToList();
+            var parsedDependencies = new Dictionary<string, List<string>>();
 
-            MoveToFront(generatorNames, new[] { "Rhetos.Dom.DomGenerator", "Rhetos.Deployment.ResourcesGenerator" } );
+            if (_buildOptions.AdditionalGeneratorDependencies == null || _buildOptions.AdditionalGeneratorDependencies.Count() == 0)
+                return parsedDependencies;
 
-            var dependencies = generators.Where(gen => gen.Dependencies != null)
-                .SelectMany(gen => gen.Dependencies.Select(dependsOn => Tuple.Create(dependsOn, GetGeneratorName(gen))))
-                .ToList();
-            Graph.TopologicalSort(generatorNames, dependencies);
+            var pairs = new List<(string name, string dependency)>();
+            foreach (var entry in _buildOptions.AdditionalGeneratorDependencies)
+            {
+                var parts = entry.Split(new[] {':'}, StringSplitOptions.RemoveEmptyEntries);
+                if (parts.Length != 2)
+                    throw new InvalidOperationException($"Invalid entry '{entry}' in {nameof(BuildOptions)}.{nameof(BuildOptions.AdditionalGeneratorDependencies)} configuration key."
+                        + " Expected \"<GeneratorName>:<GeneratorDependencyName>\" format.");
 
-            foreach (var missingDependency in dependencies.Where(dep => !generatorNames.Contains(dep.Item1)))
-                _logger.Warning($"Missing dependency '{missingDependency.Item1}' for application generator '{missingDependency.Item2}'.");

```

Comment:
What does this mean exactly? Entire design should be changed so dependencies are not really dependencies but execution order preferences? We need to rename them then, since dependency is misnomer.

Reply:
Added.